src/components/AddForm.js

```diff
@@ -12,35 +12,41 @@ function AddForm() {
   const [activitiesInput, setActivitiesInput] = useState("");
   const [locationsInput, setLocationsInput] = useState("");
   const [notesInput, setNotesInput] = useState("");
-  const [imageUpload, setImageUpload] = useState([]);
+  const [imageUploads, setImageUploads] = useState([]);
   const [imgPreview, setImgPreview] = useState([]);
   const history = useHistory();
 
   function handleSubmit(e) {
     e.preventDefault();
-    const formData = new FormData();
-    formData.append("file", imageUpload);
-    formData.append("upload_preset", "tyikvr8a");
+    const fileListAsArray = Array.from(imageUploads);
+    const imagesPromises = fileListAsArray.map((imageUpload) => {
+      const formData = new FormData();
 
-    fetch("https://api.cloudinary.com/v1_1/dyjecx1wm/image/upload", {
-      method: "PUT",
-      body: formData,
-    })
-      .then((response) => response.json())
-      .then((data) => {
-        const imageURL = data.secure_url;
+      formData.append("file", imageUpload);
+      formData.append("upload_preset", "tyikvr8a");
 
-        addTripsToLocalStorage({
-          id: destinationInput.split(" ").join("-"),
-          destination: destinationInput,
-          dates: datesInput,
-          activities: activitiesInput,
-          locations: locationsInput,
-          notes: notesInput,
-          photo: imageURL,
-        });
-        history.push("/trips");
+      return fetch("https://api.cloudinary.com/v1_1/dyjecx1wm/image/upload", {
+        method: "PUT",
+        body: formData,
+      }).then((response) => response.json());
+    });
+
+    Promise.all([...imagesPromises]).then(([...imagesResults]) => {
+      const imageURLs = imagesResults.map(
+        (imageResult) => imageResult.secure_url
+      );
+
+      addTripsToLocalStorage({
+        id: destinationInput.split(" ").join("-"),
+        destination: destinationInput,
+        dates: datesInput,
+        activities: activitiesInput,
+        locations: locationsInput,
+        notes: notesInput,
+        photos: imageURLs,
       });
+      history.push("/trips");
+    });
```

Comment:
You might want to add `.catch()` here since one of those promises might fail, which will result in rejection of the promise returned by `Promise.all()`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch